daal4py/sklearn/decomposition/_pca_0_23.py

```diff
@@ -486,7 +486,7 @@ def transform(self, X):
             if self.mean_ is not None:
                 X = X - self.mean_
                 X_transformed = np.dot(X, self.components_.T)
```

Comment:
This is a mistake. It should be like this:
```python
        if self.mean_ is not None:
            X = X - self.mean_
        X_transformed = np.dot(X, self.components_.T)
        if self.whiten:
            X_transformed /= np.sqrt(self.explained_variance_)
        return X_transformed
```
Now there is a chance that you will return None. This is bad.